ydb/core/tablet_flat/flat_table_observer.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include "defs.h"
+#include "flat_row_eggs.h"
+#include "flat_update_op.h"
+
+#include <util/generic/ptr.h>
+
+namespace NKikimr::NTable {
+
+    class ITableObserver : public TThrRefBase {
+    public:
+        /**
+         * Called when a new update is applied to the table
+         */
+        virtual void OnUpdate(
+            ERowOp rop,
+            TArrayRef<const TRawTypeValue> key,
+            TArrayRef<const TUpdateOp> ops,
+            TRowVersion rowVersion) = 0;
+
+        /**
+         * Called when an uncommitted update is applied to the table
+         */
+        virtual void OnUpdateTx(
+            ERowOp rop,
+            TArrayRef<const TRawTypeValue> key,
+            TArrayRef<const TUpdateOp> ops,
+            ui64 txId) = 0;
+    };
+
+    /**
+     * Smart pointer that can safely be used to call methods even when it's nullptr
+     */
+    class ITableObserverPtr : public TIntrusivePtr<ITableObserver> {
```

Comment:
Do we really need that wrapper? It seems that just checking 

```
if (TableObserver) {
    TableObserver->OnUpdate(rop, key, ops, rowVersion);
}
```

is easier

Reply:
Also it seems that someone may call `TableObserver->OnUpdate(rop, key, ops, rowVersion)` even now with TableObserver=nullptr